validphys2/src/validphys/theorycovariance/output.py

```diff
@@ -280,4 +275,101 @@ def plot_diag_cov_comparison(
     )
     ax.legend(fontsize=20)
     ax.margins(x=0)
+    return fig
+
+theory_covmat_custom_dataspecs = collect(theory_covmat_custom, ("dataspecs",) )
+
+@figure
+def plot_diag_cov_comparison_by_process(
+    theory_covmat_custom_dataspecs, procs_covmat, procs_data_values, dataspecs
```

Comment:
Since this function has very specific requirements for what the `dataspecs` dictionary should be, and since it should be constructed by the user in the runcard, please explain this in the docstring. Also, since it's specific to these plotting functions, it would be better if instead of `datasepecs` it got a name that makes clear that this is the case, or alternatively, avoid forcing this dictionary structure and only use top-level keys.

Reply:
Actually, as far as I can tell, `dataspecs` here is used as in the other cases documented in the code, but for the `plot_total` entry. I believe that it is probably enough to document that option then.

Reply:
[e503116](https://github.com/NNPDF/nnpdf/pull/1910/commits/e5031165edbd10305570fa22253862bfb90f1fec)